idx/metric_definition.go

```diff
@@ -19,13 +20,31 @@ const (
 	MTypeTimestamp
 )
 
+var (
+	stringBuilders = sync.Pool{New: func() interface{} { return &strings.Builder{} }}
+)
+
 //go:generate msgp
 type MetricName struct {
 	nodes []string
 }
 
 func (mn *MetricName) String() string {
-	return strings.Join(mn.nodes, ".")
+	if len(mn.nodes) == 0 {
+		return ""
+	}
+
+	builder := stringBuilders.Get().(*strings.Builder)
+	defer stringBuilders.Put(builder)
+	builder.Reset()
```

Comment:
i opt to reset elements before storing them into the pool.
that way, when retrieving items out of the pool (which is more common than storing, because pools get wiped every GC cycle), you don't need to reset anymore. minor difference i suppose, but still, conventions are nice.

Reply:
k, i'll do that

Reply:
https://github.com/grafana/metrictank/pull/1131/commits/e261be8eac8e970e99f68d9572c0fcb88f974b3f